### PR TITLE
bglibpy add 4.4.36

### PR DESCRIPTION
**Pull request by anilbey**

### CODE REVIEW

File: deploy/packages/bbp-packages.yaml
Comment (matz-e) [inline deploy/packages/bbp-packages.yaml]: These config files have been moved to `environments`, and should not require version numbers any longer. Could you please rebase on current `develop`?

Comment (anilbey) [reply deploy/packages/bbp-packages.yaml]: Ah thanks! This will make further updates easier.

**Comment by anilbey** [comment]
How's it now @matz-e ?

**Review by matz-e — DISMISSED** [review]
Looking good!

**Comment by anilbey** [comment]
@matz-e do you know why the deployment test build fails? I don't have permission to access it.

**Comment by matz-e** [comment]
@anilbey fails with
```
+ python scripts/bglibpy_test.py

Traceback (most recent call last):

  File "scripts/bglibpy_test.py", line 1, in <module>

    import bglibpy

  File "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/pulls/1054/deploy/applications/2021-02-18/linux-rhel7-x86_64/gcc-9.3.0/py-bglibpy-4.4.27-5ktphr/lib/python3.8/site-packages/bglibpy/__init__.py", line 30, in <module>

    from .ssim import SSim  # NOQA

  File "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/pulls/1054/deploy/applications/2021-02-18/linux-rhel7-x86_64/gcc-9.3.0/py-bglibpy-4.4.27-5ktphr/lib/python3.8/site-packages/bglibpy/ssim.py", line 18, in <module>

    from methodtools import lru_cache

ModuleNotFoundError: No module named 'methodtools'

script returned exit code 1
```

**Comment by anilbey** [comment]
Oh, right that that's the new python package introduced in the recent commit. 
`methodtools` is in the `setup.py`.
The python environment calling the test script below should install it. Does it not install the bglibpy package?
`python scripts/bglibpy_test.py`

**Comment by matz-e** [comment]
You'll have to add that package also as a dependency with a `depends_on` line

**Comment by anilbey** [comment]
Does the package need to be a spack module also?

**Comment by matz-e** [comment]
Yes

**Comment by anilbey** [comment]
But `bluepy` for example is using `pandas` and there is no py-pandas module in spack.

**Comment by matz-e** [comment]
Huh?
```
❯ spack list py-pandas
==> 1 packages.
py-pandas
```

**Comment by anilbey** [comment]
Oh, I didn't know it existed. 
As a user how do I load `py-pandas`, is it part of some archive or unstable?

**Comment by matz-e** [comment]
I think that `py-pandas` should be included in the `python-dev` module. And any software that has a module and itself depends on Pandas…

**Comment by anilbey** [comment]
Alright, now with this new update all dependencies are specified.

**Review by matz-e — DISMISSED** [review]

File: var/spack/repos/builtin/packages/py-bglibpy/package.py
Comment (GianlucaFicarelli) [inline var/spack/repos/builtin/packages/py-bglibpy/package.py]: `py-bglibpy` 4.4.36 should depend on `py-bluepy` >= 2.1, as in `setup.py`.
Shouldn't the version numbers be added to the dependencies?

Comment (anilbey) [reply var/spack/repos/builtin/packages/py-bglibpy/package.py]: Thanks @GianlucaFicarelli, good point.
`class PyBglibpy(PythonPackage)` has multiple versions on it.
When I specify:
`depends_on('py-bluepy@2.1', type='run')`
Is it only going to affect the latest version or is it going to effect the older versions as well?

Comment (GianlucaFicarelli) [reply var/spack/repos/builtin/packages/py-bglibpy/package.py]: If you need to keep the older versions you can add a `when` clause to specify that the dependency depends on the version.
This is an example (maybe not completely correct), but you could also take a look at `py-bluepy/package.py` to see how complicated it can become :D

```
    depends_on('py-bluepy@:1.999', type='run', when='@:4.4.35')
    depends_on('py-bluepy@2.1:', type='run', when='@4.4.36:')
```

Comment (anilbey) [reply var/spack/repos/builtin/packages/py-bglibpy/package.py]: I see, thanks! It can get complicated indeed :D

File: var/spack/repos/builtin/packages/py-bglibpy/package.py
Comment (GianlucaFicarelli) [inline var/spack/repos/builtin/packages/py-bglibpy/package.py:17]: I see that 4.4.20 has been removed. Is it done on purpose?

File: var/spack/repos/builtin/packages/py-bglibpy/package.py
Comment (GianlucaFicarelli) [inline var/spack/repos/builtin/packages/py-bglibpy/package.py:39]: iirc the upper and lower bounds are included in spack.
In that case, is it ok to use bluepy 2.0 with bglibpy up to 4.4.20? Or should it be something like 1.999?

File: var/spack/repos/builtin/packages/py-bglibpy/package.py
Comment (anilbey) [reply var/spack/repos/builtin/packages/py-bglibpy/package.py:17]: That was a mistake. Let me recover that!

File: var/spack/repos/builtin/packages/py-bglibpy/package.py
Comment (anilbey) [reply var/spack/repos/builtin/packages/py-bglibpy/package.py:39]: Yes, it's ok to use bglibpy 4.4.20 with bluepy 2.0.

**Comment by anilbey** [comment]
The error says: `py-bglibpy requires py-bluepy version 2.1:, but spec asked for 0.16.0`
How's the spec asking for 0.16.0?
Btw. bluepy 2.1.0 is not on spack yet. Might that be the cause of this error?

**Comment by matz-e** [comment]
> Btw. bluepy 2.1.0 is not on spack yet. Might that be the cause of this error?

Most likely. Why add that dependency if it's not yet in Spack?

**Comment by matz-e** [comment]
> > Btw. bluepy 2.1.0 is not on spack yet. Might that be the cause of this error?
> 
> Most likely. Why add that dependency if it's not yet in Spack?

I'll answer that myself… silly me… since this depends on #1055, I propose wait until that is merged.

**Comment by anilbey** [comment]
Haha no worries. Ok, good idea!

**Comment by anilbey** [comment]
@matz-e bluepy==2.1.0 change is merged. Can we restart the tests for this PR?

**Comment by matz-e** [comment]
@anilbey I think pushing a rebase would do that.

**Review by matz-e — CHANGES_REQUESTED** [review]
This introduces a conflict with `psp-validation`, which has
```
    depends_on('py-bglibpy@4.3.15:4.999', type='run')
    depends_on('py-bluepy@0.16.0:1.999', type='run')
```
Either please
* bump the major version of `bglibpy`
* fix the constraint above and similar ones

**Comment by GianlucaFicarelli** [comment]
@anilbey unless it makes sense to bump `bglibpy` to `5.0.0`, I can push here (in this PR) a change to the constraints for:
- `psp-validation` (the changes will be superseded by https://github.com/BlueBrain/spack/pull/1074 after merging this PR)
- `py-minis-validation`

I've seen that also `py-sonata-network-reduction` depends on `py-bglibpy@4.3:`, but it doesn't require `py-bluepy` so I think that no changes are needed in this case.

**Review by matz-e — APPROVED** [review]

**Comment by GianlucaFicarelli** [comment]
@wvangeit the CI is finally passing (thanks @matz-e), but I think @anilbey is off today.
Could you please confirm if we can merge this PR, so I can proceed to rebase https://github.com/BlueBrain/spack/pull/1074?

**Comment by wvangeit** [comment]
Yes, it can be merged

**Pull request merged**
